### PR TITLE
Increased coverage and added bug fixing

### PR DESCRIPTION
**Pull request by umbreak**
Partially addresses #367

### CODE REVIEW

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
   * A value with an attached revision corresponding to the value.
   *
   * @param rev   the value revision
   * @param value the value
   */
 final case class RevisionedValue[A](rev: Long, value: A) extends RegisteredValue[A]
-
-object RevisionedValue {
-
-  private[async] def revisionedValueClock[A]: Clock[RevisionedValue[A]] =
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala]: Why the removal?

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala [hunk 2]
```diff
@@ -1,17 +1,9 @@
...
 
-import akka.cluster.ddata.LWWRegister.Clock
-
 /**
   * A value with an attached revision corresponding to the value.
   *
   * @param rev   the value revision
   * @param value the value
   */
 final case class RevisionedValue[A](rev: Long, value: A) extends RegisteredValue[A]
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala]: I would recommend renaming `RegisteredValue[A]` to `RegisterValue[A]`.

Comment (bogdanromanx) [reply src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala]: And possibly make it a sealed trait while moving the sub types in the same file.

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/ResolverEncoder.scala [hunk 3]
```diff
@@ -1,21 +1,29 @@
 package ch.epfl.bluebrain.nexus.kg.resolve
 
+import ch.epfl.bluebrain.nexus.commons.types.search.{QueryResult, QueryResults}
 import ch.epfl.bluebrain.nexus.iam.client.types.Identity
 import ch.epfl.bluebrain.nexus.iam.client.types.Identity.{AuthenticatedRef, GroupRef, UserRef}
+import ch.epfl.bluebrain.nexus.kg.config.Contexts.{resolverCtx, _}
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/resolve/ResolverEncoder.scala]: fix import

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Ref.scala [hunk 4]
```diff
@@ -41,10 +41,13 @@ object Ref {
...
         val (q, opt) = extractTagRev(query)
-        refOf(u.copy(query = Some(q)), opt)
+        val qry      = if (q.value.isEmpty) None else Some(q)
+        refOf(u.copy(query = qry), opt)
       case u @ Urn(_, _, _, Some(query), _) =>
         val (q, opt) = extractTagRev(query)
-        refOf(u.copy(q = Some(q)), opt)
+        val qry      = if (q.value.isEmpty) None else Some(q)
+        refOf(u.copy(q = qry), opt)
+
```
Comment (bogdanromanx) [inline src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Ref.scala]: empty line

File: src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala (hunk 1 above)
Comment (umbreak) [reply src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala]: It wasn't used. When we use it I will add it, but right now is just garbage

Comment (bogdanromanx) [reply src/main/scala/ch/epfl/bluebrain/nexus/kg/async/RevisionedValue.scala]: It is used, but not immediately obvious as creating a register takes a default clock implicitly:
```
/**
   * INTERNAL API
   */
  @InternalApi private[akka] def apply[A](node: UniqueAddress, initialValue: A, clock: Clock[A]): LWWRegister[A] =
    new LWWRegister(node, initialValue, clock(0L, initialValue))

  def apply[A](initialValue: A)(implicit node: Cluster, clock: Clock[A] = defaultClock[A]): LWWRegister[A] =
    apply(node.selfUniqueAddress, initialValue, clock)
```

Also, the only reason this type exists is to be used in a LWWRegister; if the clock is "garbage" the whole type is "garbage".

**Pull request merged**